plugins/osdn/osdn.go

```diff
@@ -92,6 +114,56 @@ func (oi *OsdnRegistryInterface) WatchSubnets(receiver chan<- *osdnapi.SubnetEve
 	}
 }
 
+func newSDNPod(kPod *kapi.Pod) osdnapi.Pod {
+	containerID := ""
+	if len(kPod.Status.ContainerStatuses) > 0 {
+		// Extract only container ID, pod.Status.ContainerStatuses[0].ContainerID is of the format: docker://<containerID>
+		containerID = strings.Split(kPod.Status.ContainerStatuses[0].ContainerID, "://")[1]
+	}
+	return osdnapi.Pod{
+		Name:        kPod.ObjectMeta.Name,
+		Namespace:   kPod.ObjectMeta.Namespace,
+		ContainerID: containerID,
+	}
+}
+
+func (oi *OsdnRegistryInterface) GetPods() ([]osdnapi.Pod, string, error) {
+	kPodList, err := oi.kClient.Pods(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return nil, "", err
+	}
+
+	oPodList := make([]osdnapi.Pod, 0, len(kPodList.Items))
+	for _, kPod := range kPodList.Items {
+		if kPod.Status.PodIP != "" {
+			oi.namespaceOfPodIP[kPod.Status.PodIP] = kPod.ObjectMeta.Namespace
+		}
+		oPodList = append(oPodList, newSDNPod(&kPod))
+	}
+	return oPodList, kPodList.ListMeta.ResourceVersion, nil
+}
+
+func (oi *OsdnRegistryInterface) WatchPods(ready chan<- bool, start <-chan string, stop <-chan bool) error {
+	eventQueue, startVersion := oi.createAndRunEventQueue("Pod", ready, start)
+
+	checkCondition := true
+	for {
+		eventType, obj, err := getEvent(eventQueue, startVersion, &checkCondition)
+		if err != nil {
+			return err
+		}
+		kPod := obj.(*kapi.Pod)
+
+		switch eventType {
+		case watch.Added:
```

Comment:
oops, yeah. (well, the code works as is, so I guess that means the IP is always added after initial creation)